docs/2_3_common_states.adoc

```diff
@@ -82,485 +87,644 @@ If `loggingOn == fmi3True`, the FMU enables a vendor defined set of `<LogCategor
-[#figure-events]
-image::images/Event.svg[width=60%, align="center"]
+<<EventMode>> is only available in Model Exchange and Co-Simulation.
+In <<EventMode>> all continuous-time, discrete-time equations and active model partitions are evaluated.
 
-[start=4]
-. At every completed step of an integrator, <<fmi3CompletedIntegratorStep>> must be called (provided the capability flag `completedIntegratorStepNotNeeded` of `<fmiModelDescription>` is `false`).
-An event occurs at this time instant, if indicated by the return argument `enterEventMode == fmi3True`.
-Such an event is called <<step event>>.
-_[<<step event,`Step events`>> are, for example, used to dynamically change the (continuous) <<state,`states`>> of a model internally in the FMU, because the previous states are no longer suited numerically.]_
+There are multiple kinds of events that require a transition to <<EventMode>>:
+
+* <<time-event,Time events>>: The time instant of such an event is signaled to the importer by
+  - the argument <<nextEventTime>> of <<fmi3UpdateDiscreteStates>>, or
+  - the arguments of <<fmi3GetShift>> and <<fmi3GetInterval>>.
+* <<state-event,State events>>
+* <<input-event,Input events>>
+* <<step-event,Step events>>
+* Co-Simulation: <<fmi3DoStep>> returns with <<eventEncountered,`eventEncountered == fmi3True`>> or <<clocksAboutToTick,`clocksAboutToTick == fmi3True`>>.
+
+[#table-math-event-mode]
+[cols="5,3",options="header",]
+|====
+|Equations and Actions
+|Functions Influencing Equations
+
+|Set <<tunable>> <<parameter,`parameters`>> latexmath:[\mathbf{p}_{\mathit{variability == tunable}}]
+|`fmi3Set{VariableType}`
+
+|Set continuous-time and discrete-time <<input,`inputs`>> latexmath:[\mathbf{u}_{c+d}(\mathbf{t})]
+|`fmi3Set{VariableType}`
 
-In this mode all continuous-time and discrete-time equations are active and the unknowns at an event can be computed and retrieved.
-The event time of a <<state event>> may be determined if a domain change of at least one event indicator is detected at the end of a completed integrator step.
+a|
+* Activate clocks and respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}]
+* Deactivate clocks and
+  - Reset respective clocked variables to their previous values: latexmath:[\mathbf{v}_{\mathit{d}} := {}^\bullet\mathbf{v}_{\mathit{d}}]
+  - Deactivate respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}]
+|<<fmi3SetClock>>
+
+a|
+* Activate output clocks and respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}] +
+* latexmath:[(\mathbf{y}_{c+d}, \mathbf{\dot{x}}_c, \mathbf{x}_{c+d}, \mathbf{z}, \mathbf{r}, \mathbf{w}_{c+d}, \mathbf{b}, \mathbf{T}_{\mathit{next}}, \mathbf{T}_\mathit{interval}) := \mathbf{f}_{\mathit{event}}({}^\bullet\mathbf{x}_{c+d}, \mathbf{u}_{c+d}, \mathbf{p}, {}^\bullet\mathbf{b}, \mathbf{t})]
+|<<fmi3GetClock>>, +
+<<fmi3GetInterval>>, +
+`fmi3Get{VariableType}`, +
```

Comment:
Is the integer part of super dense time incremented by such an fmi3Get{VariablesType} call?

Reply:
No, the time is only updated in fmi3UpdateDiscreteStates - as you see in its line of the table.